File: lib/item.rb
class Item
    attr_reader :name, :price, :quantity

    FOODS = ['chocolate', 'chocolates']
    MEDICALS = ['headache pills']
    BOOKS = ['book']

    def initialize(name, price, quantity)
        @name = name
        @price = price
        @quantity = quantity
    end

    def imported?
        name.downcase.include?("imported")
    end

    def taxable?
        lower_case_name = name.downcase
        exempt = FOODS + MEDICALS + BOOKS
        !exempt.any? { |w| lower_case_name.match?("\\b#{Regexp.escape(w)}\\b") }
    end
end

File: lib/item.py
import re


class Item:
    FOODS = ['chocolate', 'chocolates']
    MEDICALS = ['headache pills']
    BOOKS = ['book']

    def __init__(self, name, price, quantity):
        self.name = name
        self.price = price
        self.quantity = quantity

    def imported(self):
        return 'imported' in self.name.lower()

    def taxable(self):
        lower_case_name = self.name.lower()
        exempt = self.FOODS + self.MEDICALS + self.BOOKS
        return not any(re.search(r'\b' + re.escape(w) + r'\b', lower_case_name) for w in exempt)
